### PR TITLE
[ESP32] doc:added doc for vs code development

### PR DESCRIPTION
**Pull request by sayondeep**
## Visual Studio Code Setup Guide for Matter Development on ESP32 devices

This pull request adds comprehensive instructions for setting up Visual Studio Code for Matter development on esp32 devices for both Linux and Windows systems. The guide covers the necessary steps to configure the development environment, connect to devices, and build applications.

### Changes Made

- Added step-by-step instructions for Linux setup, including connecting Espressif esp32 devices, configuring the `devcontainer.json` file, installing `esp-idf`, activating Matter, and building projects.
- Added Windows setup steps, including connecting to WSL, cloning the Matter repository, setting up serial port access, 
- The guide has been tested on Windows 11 Pro with WSL 2 and Ubuntu 22.04 LTS, as well as Ubuntu 22.04 LTS.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: docs/guides/esp32/vs_code_development.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Visual Studio Code Development
```
Comment (dhrishi) [inline docs/guides/esp32/vs_code_development.md:1]: Should there be an entry for this in: https://github.com/project-chip/connectedhomeip/tree/master/docs/guides/esp32#espressif-esp32-getting-started-guide ?

File: docs/guides/esp32/vs_code_development.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+[these](https://github.com/project-chip/connectedhomeip/edit/master/docs/VSCODE_DEVELOPMENT.md)
+steps to set up Visual Studio Code for Matter development.
+
+### Linux <a name="linux">
+
+For Espressif esp32 devices:
+
+1. Connect your device to the system before running the container.
+
+1. Add `"--device=/dev/ttyUSB0",` inside runArgs in
```
Comment (dhrishi) [inline docs/guides/esp32/vs_code_development.md:27]: Like `/dev/ttyUSB0` on Linux or `/dev/tty.usbserial-101` on MacOS ?

File: docs/guides/esp32/vs_code_development.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+   window you should have a new box prompting you to re-open the window as a
+   container. Hit yes.
+
+1. The container will be built.
+
+1. Install esp-idf : `cd /opt/espressif/esp-idf && ./install.sh`
+
+1. Source esp-idf : `source /opt/espressif/esp-idf/export.sh`
+
+1. Activate matter :
```
Comment (dhrishi) [inline docs/guides/esp32/vs_code_development.md:41]: Matter, at all the places

File: docs/guides/esp32/vs_code_development.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Add `"--device=/dev/ttyUSB0",` inside runArgs in
+   [devcontainer.json](../../../.devcontainer/devcontainer.json). Confirm the
+   port of the device and change the parameter accordingly.
+
+1. On opening connectedhomeip in Visual Studio code, At the bottom right of your
+   window you should have a new box prompting you to re-open the window as a
+   container. Hit yes.
+
+1. The container will be built.
```
Comment (dhrishi) [inline docs/guides/esp32/vs_code_development.md:35]: Why is the same number "1." seen?

Comment (sayondeep) [reply docs/guides/esp32/vs_code_development.md:35]: Then it is automatically numbered.

File: docs/guides/esp32/vs_code_development.md (hunk 1 above)
Comment (sayondeep) [reply docs/guides/esp32/vs_code_development.md:1]: Yes we should add it there

File: docs/guides/esp32/vs_code_development.md (hunk 2 above)
Comment (sayondeep) [reply docs/guides/esp32/vs_code_development.md:27]: Yes . Correct

**Review by dhrishi — APPROVED** [review]

**Pull request merged**